libcontainer/nsenter/nsexec.c

```diff
@@ -597,8 +651,8 @@ void nsexec(void)
 						update_setgroups(child, SETGROUPS_DENY);
 
 					/* Set up mappings. */
-					update_uidmap(child, config.uidmap, config.uidmap_len);
-					update_gidmap(child, config.gidmap, config.gidmap_len);
+					update_uidmap(child, config.is_rootless, config.uidmap, config.uidmap_len);
+					update_gidmap(child, config.is_rootless, config.gidmap, config.gidmap_len);
```

Comment:
Why not always try to use `newuidmap` and `newgidmap` rather than making it conditional on `is_rootless`? (Though maybe you're right here.)